worldedit-core/src/main/java/com/fastasyncworldedit/core/function/pattern/RelativePattern.java

```diff
@@ -63,4 +63,11 @@ public void reset() {
         origin = null;
     }
 
+    @Override
+    public Pattern fork() {
+        RelativePattern forked = new RelativePattern(this.pattern.fork(), this.minY, this.maxY);
+        forked.origin = this.origin; // maintain origin for forks
```

Comment:
Should probably look at having the origin be `toImmutable()`ised when it's set

Reply:
That would otherwise already cause bugs I think? But yeah, that might make sense.

Reply:
Possibly, but can't be too careful aha